x-pack/plugins/ml/public/application/components/annotations/annotations_table/annotations_table.js

```diff
@@ -447,38 +482,116 @@ export class AnnotationsTable extends Component {
       });
     }
 
-    columns.push({
-      align: RIGHT_ALIGNMENT,
-      width: '60px',
-      name: i18n.translate('xpack.ml.annotationsTable.actionsColumnName', {
-        defaultMessage: 'Actions',
-      }),
-      actions,
-    });
-
     const getRowProps = (item) => {
       return {
         onMouseOver: () => this.onMouseOverRow(item),
         onMouseLeave: () => this.onMouseLeaveRow(),
       };
     };
+    let filterOptions = [];
+    if (this.props.aggregations?.event?.buckets) {
+      const buckets = this.props.aggregations.event.buckets;
+      const foundUser = buckets.findIndex((d) => d.key === ANNOTATION_EVENT_USER) > -1;
+      filterOptions = foundUser
+        ? buckets
+        : [{ key: ANNOTATION_EVENT_USER, doc_count: 0 }, ...buckets];
+    }
+    const filters = [
+      {
+        type: 'field_value_selection',
+        field: 'event',
+        name: 'Event',
+        multiSelect: 'or',
+        options: filterOptions.map((field) => ({
+          value: field.key,
+          name: field.key,
+          view: `${field.key} (${field.doc_count})`,
+        })),
+      },
+    ];
+
+    if (
+      this.props.chartDetails?.entityData?.entities &&
+      this.props.chartDetails?.entityData?.entities.length > 0
+    ) {
+      this.props.chartDetails?.entityData?.entities.forEach((entity) => {
+        if (entity.fieldType === 'partition_field') {
+          columns.push({
+            field: 'partition_field_value',
+            name: i18n.translate('xpack.ml.annotationsTable.partitionColumnName', {
+              defaultMessage: 'Partition',
+            }),
+            sortable: true,
+          });
+        }
+
+        if (entity.fieldType === 'over_field') {
+          columns.push({
+            field: 'over_field_value',
+            name: i18n.translate('xpack.ml.annotationsTable.overColumnName', {
+              defaultMessage: 'Over',
+            }),
+            sortable: true,
+          });
+        }
+        if (entity.fieldType === 'by_field') {
+          columns.push({
+            field: 'by_field_value',
+            name: i18n.translate('xpack.ml.annotationsTable.byColumnName', {
+              defaultMessage: 'By',
+            }),
+            sortable: true,
+          });
+        }
+      });
+
+      filters.push({
+        type: 'is',
+        field: CURRENT_SERIES,
+        name: i18n.translate('xpack.ml.annotationsTable.seriesOnlyFilterName', {
+          defaultMessage: 'Filter to series',
+        }),
+      });
+    }
+    const search = {
+      defaultQuery: queryText,
+      box: {
+        incremental: true,
+        schema: true,
+      },
+      filters: filters,
+    };
+
+    columns.push(
+      {
+        align: RIGHT_ALIGNMENT,
+        width: '60px',
+        name: i18n.translate('xpack.ml.annotationsTable.actionsColumnName', {
+          defaultMessage: 'Actions',
```

Comment:
For the link to the Single Metric Viewer from the table in the Anomaly Explorer, if the annotation is at the series level, it needs to pass the partitioning field(s) and detector index to the Single Metric Viewer:

![image](https://user-images.githubusercontent.com/7405507/86463425-5ffbd280-bd25-11ea-9d80-f8e435a11b9c.png)

Reply:
Added that functionality here https://github.com/elastic/kibana/pull/70198/commits/1f2684dbbc91ca4a7d83ab93d8d4cfb79a5180c2

![annotations_2020-07-08 at 4 28 PM](https://user-images.githubusercontent.com/43350163/86975759-9f1f8e80-c13e-11ea-85e0-2fd9baea55da.gif)
![annotations_2020-07-08 at 4 29 PM](https://user-images.githubusercontent.com/43350163/86975769-a5ae0600-c13e-11ea-9367-2512afedd829.gif)